students/aravichander/lesson06/Mailroom3.py

```diff
@@ -0,0 +1,89 @@
+#Mailroom #2
+
+import sys
+import json
+import unittest 
+
+Donors = {"Bill Gates":[10000,12000],
+		  "Jeff Bezos":[50],`
+		  "Mark Zuckerberg": [500,600,700],
+		  "Paul Allen": [250,350,450],
+		  "King of Siam": [200,250]
+		  }
+
+def printdonors():
+	for x,v in Donors.items():
+		print(x,"donated $",v)
+
+def thankall():
+	for x in Donors:
+		totaldonated = sum(Donors[x])
+		letterofthanks(x,totaldonated)
+
+def letterofthanks(donorname,totaldonated):
+	#To add sum of donations
+	filename = donorname+".txt"
+	f = open(filename,"w+")
+	statementtowrite = "Thank you "+donorname+" for your donation of $"+str(totaldonated)+". We greatly appreciate your help."
+	f.write(statementtowrite)
+	f.close()
+	print("\nLetter of thanks has been saved to hard disk as text file under "+str(filename))
+
+def searchdonors(thankdonor):
+	#Not sure if these exceptions here are needed to be honest#
+	for x in Donors:
+		try:
+			if thankdonor == x:
+				totaldonated = sum(Donors[x])
+				letterofthanks(thankdonor,totaldonated)
+		except KeyError:
```

Comment:
you are right -- not helpful here -- the key is gotten from the for loop, so it is in the dict, and you won't ever get a KeyError.